File: model/account_thresholds.go
package model

import (
	b64 "encoding/base64"
  "github.com/stellar/go/xdr"
)

func NewAccountThresholdsFromRaw(raw string, id string) AccountThresholds {
	t, err := b64.StdEncoding.DecodeString(raw)
	if (err != nil) { return AccountThresholds{} }

	return AccountThresholds{
		ID: id,
		MasterWeight: int(t[xdr.ThresholdIndexesThresholdMasterWeight]),
		Low: int(t[xdr.ThresholdIndexesThresholdLow]),
		Medium: int(t[xdr.ThresholdIndexesThresholdMed]),
		High: int(t[xdr.ThresholdIndexesThresholdHigh]),
	}
}

File: src/model/account_thresholds.ts
export interface IAccountThresholds {
  masterWeight: number;
  low: number;
  medium: number;
  high: number;
}

export class AccountThresholds implements IAccountThresholds {
  public masterWeight: number;
  public low: number;
  public medium: number;
  public high: number;

  constructor(data: IAccountThresholds) {
    this.masterWeight = data.masterWeight;
    this.low = data.low;
    this.medium = data.medium;
    this.high = data.high;
  }

  public equals(other: IAccountThresholds): boolean {
    return (
      this.low === other.low &&
      this.medium === other.medium &&
      this.high === other.high &&
      this.masterWeight === other.masterWeight
    );
  }
}
